src/StreamTarget.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Log;
+
+use InvalidArgumentException;
+use RuntimeException;
+
+use function error_get_last;
+use function fclose;
+use function flock;
+use function fopen;
+use function fwrite;
+use function gettype;
+use function get_resource_type;
+use function is_resource;
+use function sprintf;
+use function stream_get_meta_data;
+
+use const LOCK_EX;
+use const LOCK_UN;
+
+/**
+ * StreamTarget is the log target that writes to the specified output stream.
+ */
+final class StreamTarget extends Target
+{
+    /**
+     * @var resource|string The string stream identifier or stream resource.
+     *
+     * @psalm-var mixed
+     */
+    private $stream;
+
+    /**
+     * @param resource|string $stream The string stream identifier or stream resource.
+     */
+    public function __construct($stream = 'php://stdout')
+    {
+        $this->stream = $stream;
+        parent::__construct();
+    }
+
+    protected function export(): void
+    {
+        $stream = $this->createStream();
+        flock($stream, LOCK_EX);
+
+        if (fwrite($stream, $this->formatMessages("\n")) === false) {
+            fclose($stream);
```

Comment:
Shouldn't we do `flock($stream, LOCK_UN);` here?

Reply:
Do: https://github.com/yiisoft/log/pull/55/files#diff-1e808bd6ab1e9770827aff7f2022823cf221774f5e6859401d182a109db7aaf9R59